Marlin/src/core/utility.h

```diff
@@ -33,7 +33,7 @@ void safe_delay(millis_t ms);           // Delay ensuring that temperatures are
   inline void serial_delay(const millis_t) {}
 #endif
 
-#if GRID_MAX_POINTS
+#if TERN(PROUI_GRID_PNTS, HAS_MESH, GRID_MAX_POINTS)
```

Comment:
You can't use `TERN()` with the integer define `PROUI_GRID_PNTS`. Instead:
```cpp
#if (PROUI_GRID_PNTS > 0 && HAS_MESH) || (PROUI_GRID_PNTS == 0 && GRID_MAX_POINTS > 0)
```

Reply:
I think you may be mixing up that with `hmiData.grid_max_points`, which is the variable integer. `PROUI_GRID_PNTS` simply allows this feature to be used.

I also want to make a note, mostly regarding your recent commit to fix compiling error. so I haven't been able to actually get the variable/customizable `MESH_INSET` or more specifically `MESH_MAX(MIN)_X(Y)` to work properly. in my testing, yes I was able to at least get the values to register and change, but not implement itself when building a mesh. for instance, when editing the values and say setting the mesh to a small square area, it just didnt apply, even though the values were saved and remained after restart.

I just hope I have done it right. you see, I looked through MRiscoC's source code which has the same feature, except some of it is behind the inaccessible library. I might have to completely omit the editable mesh_inset part, because I know the grid points and the other things do work.

any idea what can be done so changing the mesh_inset can be done through the UI?

Reply:
it comes down to the function `get_mesh_x/y()`

so for instance with UBL, this simply did not work:
```y
    static float get_mesh_x(const uint8_t i) {
      return i < (GRID_MAX_POINTS_X) ? pgm_read_float(&_mesh_index_to_xpos[i]) : MESH_MIN_X + i * (MESH_X_DIST);
    }
```

but I got it to work with this (at least for UBL):
```y
    static float get_mesh_x(const uint8_t i) {
      return MESH_MIN_X + i * (MESH_X_DIST);
    }
    static float get_mesh_y(const uint8_t j) {
      return MESH_MIN_Y + j * (MESH_Y_DIST);
    }
```